src/constraints/constraints_helper.jl

```diff
@@ -0,0 +1,454 @@
+export
+    DistanceConstraint,
+    AngleConstraint,
+    disable_constrained_interactions!,
+    apply_position_constraints!,
+    apply_velocity_constraints!
+
+
+abstract type Constraint{D} end
+
+"""
+    DistanceConstraint(i, j, dist)
+
+
+    DistanceConstraint(i, j, dist)
+
+Constraint between two atoms that maintains a fixed distance between the two atoms.
+"""
+struct DistanceConstraint{D} <: Constraint{D}
```

Comment:
Relaxing the integer type from `Int` to a parametric type `I` means it can take `Int32`, which makes setup from bonds easier. Allowing `Int32` throughout might help GPU performance a bit.